### PR TITLE
TASK: Rename hostPattern to hostname in Domain

### PR DESCRIPTION
**Pull request by kdambekalns**
Since 2013 the `hostPattern` has a `HostnameValidator` attached, so setting
a wildcard was not supported any longer.

This change renames `hostPattern` to `hostname` throughout the code and
adjusts the `DomainMatchingStrategy` to no longer support the `*` wildcard.
It therefore makes things clearer and avoid misconceptions about how
Domain entries work in Neos.

### CODE REVIEW

File: TYPO3.Neos/Classes/TYPO3/Neos/Command/DomainCommandController.php [hunk 1]
```diff
@@ -47,22 +47,22 @@ class DomainCommandController extends \TYPO3\Flow\Cli\CommandController
      * Add a domain record
      *
      * @param string $siteNodeName The nodeName of the site rootNode, e.g. "neostypo3org"
-     * @param string $hostPattern The host pattern to match on, e.g. "neos.typo3.org"
+     * @param string $hostname The host name to match on, e.g. "neos.typo3.org"
```
Comment (aertmann) [inline TYPO3.Neos/Classes/TYPO3/Neos/Command/DomainCommandController.php]: hostname

File: TYPO3.Neos/Classes/TYPO3/Neos/Command/DomainCommandController.php [hunk 2]
```diff
@@ -94,115 +94,88 @@ public function addCommand($siteNodeName, $hostPattern, $scheme = null, $port =
     /**
      * Display a list of available domain records
      *
-     * @param string $hostPattern An optional host pattern to search for
+     * @param string $hostname An optional host name to search for
```
Comment (aertmann) [inline TYPO3.Neos/Classes/TYPO3/Neos/Command/DomainCommandController.php]: hostname

File: TYPO3.Neos/Classes/TYPO3/Neos/Command/DomainCommandController.php [hunk 3]
```diff
@@ -94,115 +94,88 @@ public function addCommand($siteNodeName, $hostPattern, $scheme = null, $port =
...
-                ($domain['active'] ? 'Active' : 'Inactive')
-            ));
-        }
-        $this->outputLine();
+        $this->output->outputTable($availableDomains, ['Node name', 'Domain (Scheme/Host/Port)', 'State']);
     }
 
     /**
-     * Delete a domain record
+     * Delete a host (domain) record
```
Comment (aertmann) [inline TYPO3.Neos/Classes/TYPO3/Neos/Command/DomainCommandController.php]: this doesn't make sense to me.. it's deleting a domain not a host.. could be `Delete a domain record by hostname` if it needs to be more clear

File: TYPO3.Neos/Classes/TYPO3/Neos/Command/DomainCommandController.php [hunk 4]
```diff
@@ -94,115 +94,88 @@ public function addCommand($siteNodeName, $hostPattern, $scheme = null, $port =
...
         }
 
         $this->domainRepository->remove($domain);
-        $this->outputLine('Domain deleted.');
+        $this->outputLine('Domain entry deleted.');
     }
 
     /**
-     * Activate a domain record
+     * Activate a host (domain) record
```
Comment (aertmann) [inline TYPO3.Neos/Classes/TYPO3/Neos/Command/DomainCommandController.php]: same here

File: TYPO3.Neos/Classes/TYPO3/Neos/Command/DomainCommandController.php [hunk 5]
```diff
@@ -94,115 +94,88 @@ public function addCommand($siteNodeName, $hostPattern, $scheme = null, $port =
...
 
         $domain->setActive(true);
         $this->domainRepository->update($domain);
-        $this->outputLine('Domain activated.');
+        $this->outputLine('Domain entry activated.');
     }
 
     /**
-     * Deactivate a domain record
+     * Deactivate a host (domain) record
```
Comment (aertmann) [inline TYPO3.Neos/Classes/TYPO3/Neos/Command/DomainCommandController.php]: and here

File: TYPO3.Neos/Classes/TYPO3/Neos/Domain/Model/Domain.php [hunk 6]
```diff
@@ -60,26 +62,49 @@ class Domain implements \TYPO3\Flow\Cache\CacheAwareInterface
     protected $active = true;
 
     /**
-     * Sets the pattern for the host of the domain
+     * Sets the hostname
+     *
+     * @param string $hostname
+     * @return void
```
Comment (aertmann) [inline TYPO3.Neos/Classes/TYPO3/Neos/Domain/Model/Domain.php]: missing `@api`

File: TYPO3.Neos/Classes/TYPO3/Neos/Domain/Model/Domain.php [hunk 7]
```diff
@@ -60,26 +62,49 @@ class Domain implements \TYPO3\Flow\Cache\CacheAwareInterface
...
     public function setHostPattern($hostPattern)
     {
-        $this->hostPattern = $hostPattern;
+        $this->hostname = $hostPattern;
+    }
+
+    /**
+     * Returns the hostname
+     *
+     * @return string
```
Comment (aertmann) [inline TYPO3.Neos/Classes/TYPO3/Neos/Domain/Model/Domain.php]: missing `@api`

File: TYPO3.Neos/Classes/TYPO3/Neos/Domain/Repository/DomainRepository.php [hunk 8]
```diff
@@ -39,36 +39,36 @@ class DomainRepository extends \TYPO3\Flow\Persistence\Repository
...
-        return $this->domainMatchingStrategy->getSortedMatches($host, $domains);
+        return $this->domainMatchingStrategy->getSortedMatches($hostname, $domains);
     }
 
     /**
-     * Find the best matching active domain for the given host.
+     * Find the best matching active domain for the given hostname.
      *
-     * @param string $host Host the domain should match with (eg. "localhost" or "www.neos.io")
+     * @param string $hostname Host the domain should match with (eg. "localhost" or "www.neos.io")
```
Comment (aertmann) [inline TYPO3.Neos/Classes/TYPO3/Neos/Domain/Repository/DomainRepository.php]: Hostname the domain..

File: TYPO3.Neos/Classes/TYPO3/Neos/Domain/Repository/DomainRepository.php [hunk 9]
```diff
@@ -39,36 +39,36 @@ class DomainRepository extends \TYPO3\Flow\Persistence\Repository
...
     );
 
     /**
-     * Finds all active domains matching the given host.
+     * Finds all active domains matching the given hostname.
      *
      * Their order is determined by how well they match, best match first.
      *
-     * @param string $host Host the domain should match with (eg. "localhost" or "www.neos.io")
+     * @param string $hostname Host the domain should match with (eg. "localhost" or "www.neos.io")
```
Comment (aertmann) [inline TYPO3.Neos/Classes/TYPO3/Neos/Domain/Repository/DomainRepository.php]: Hostname the domain..

File: TYPO3.Neos/Classes/TYPO3/Neos/Domain/Service/DomainMatchingStrategy.php [hunk 10]
```diff
@@ -25,29 +25,30 @@ class DomainMatchingStrategy
 
     /**
      * Returns those of the given domains which match the specified host.
+     *
      * The domains are sorted by their match exactness.
      * If none really matches an empty array is returned.
      *
-     * @param string $host The host to match against (eg. "localhost" or "www.neos.io")
+     * @param string $hostnameToMatch The host name to match against (eg. "localhost" or "www.neos.io")
```
Comment (aertmann) [inline TYPO3.Neos/Classes/TYPO3/Neos/Domain/Service/DomainMatchingStrategy.php]: hostname

**Comment by aertmann** [comment]
hey @kdambekalns thanks for taking care, left some nitpick comments

File: TYPO3.Neos/Classes/TYPO3/Neos/Command/DomainCommandController.php (hunk 3 above)
Comment (kdambekalns) [inline TYPO3.Neos/Classes/TYPO3/Neos/Command/DomainCommandController.php]: I had this change lying around for a while… but I _think_ that was to make clear it is not deleting a domain (you can only to that with your domain registrar). But… nitpicking both ways. :) I'll change that to what is was earlier.

**Pull request merged**